### ISSUE TITLE
softmax_classifier input argument not right?

### ISSUE THREAD

**Issue by weiliu620** [tensorflow/tensorflow#2330]
The softmax_classifier in losses_ops.py file seems not right to me. I haven't done any test yet, but just look at the the input argument definition, the tensor_in and weight are not compatible for matrix multiplication. 
====== begin ============
tensor_in: Input tensor, [batch_size, feature_size], features.
    labels: Tensor, [batch_size, n_classes], labels of the output classes.
    weights: Tensor, [batch_size, feature_size], linear transformation
      matrix.
    biases: Tensor, [batch_size], biases.
======== end ==============
Their inner dimensions do not match for matmul in xw_plus_b function. 

Also, since it is done in a batch, isn't it we should use batch_matmul?

**Comment by petewarden** [comment]
It would be great to see a test on this one if you're able to put something in code. Thanks!

**Comment by girving** [comment]
I'll close for now since this isn't enough information to reproduce without some example code or perhaps the details of the error.  Please comment and we can reopen if there is more information available.
